cfy_manager/components/prometheus/prometheus.py

```diff
@@ -0,0 +1,319 @@
+#     for exporter in EXPORTERS:
+#         logger.notice('Installing {0}...'.format(exporter['description']))
+#         working_dir = _create_directory(join(sep, 'tmp', exporter['name']))
+#         archive_file_name = _download_exporter(exporter, working_dir)
+#         _unpack_archive(archive_file_name, working_dir)
+#         _deploy_exporter(exporter, working_dir)
+#         logger.notice(
+#             '{0} successfully installed'.format(exporter['description']))
+
+
+def _chown_resources_dir():
+    logger.notice('Changing files and directories ownership for Prometheus')
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP,
+                 join(BIN_DIR, 'prometheus'))
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP,
+                 join(BIN_DIR, 'promtool'))
+    for exporter in EXPORTERS:
+        common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP,
+                     join(BIN_DIR, exporter['name']))
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP, PROMETHEUS_CONFIG_DIR)
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP, PROMETHEUS_DATA_DIR)
+
+
+def _deploy_configuration():
+    logger.info('Initializing Prometheus...')
+    _deploy_prometheus_configuration()
+    _deploy_exporters_configuration()
+    _deploy_services_configuration()
+    service.reload(PROMETHEUS,
+                   append_prefix=False, ignore_failure=True)
+    for exporter in EXPORTERS:
+        service.reload(exporter['name'], append_prefix=False,
+                       ignore_failure=True)
+
+
+def _deploy_prometheus_configuration():
+    logger.notice('Deploying Prometheus configuration...')
+    files.deploy(join(CONFIG_DIR, 'prometheus.yml'),
+                 PROMETHEUS_CONFIG_PATH)
+    common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP, PROMETHEUS_CONFIG_PATH)
+
+
+def _deploy_exporters_configuration():
+    for exporter in EXPORTERS:
+        if 'deploy_config' not in exporter:
+            continue
+        logger.notice(
+            'Deploying {0} configuration...'.format(exporter['description']))
+        for file_name, dest_file_name in exporter['deploy_config'].items():
+            files.deploy(join(CONFIG_DIR, file_name), dest_file_name)
+            common.chown(CLOUDIFY_USER, CLOUDIFY_GROUP, dest_file_name)
+
+
+def _deploy_services_configuration():
+    logger.notice('Adding Prometheus service configuration...')
+    files.deploy(join(CONFIG_DIR, 'prometheus.service'),
+                 join(SYSTEMD_CONFIG_DIR, 'prometheus.service'))
+    for exporter in EXPORTERS:
+        files.deploy(join(CONFIG_DIR,
+                          '{0}.service'.format(exporter['name'])),
+                     join(SYSTEMD_CONFIG_DIR,
+                          '{0}.service'.format(exporter['name'])))
+
+
+def _validate_prometheus_running():
+    logger.info('Making sure Prometheus is live...')
+    service.verify_alive(PROMETHEUS, append_prefix=False)
+
+
+@argh.arg('-v', '--verbose', help=constants.VERBOSE_HELP_MSG, default=False)
+def start(verbose=False):
```

Comment:
not sure if the start/stop cli calls are actually useful? maybe we dont need them?

Reply:
removed